pkg/controller/datadogagent/service.go

```diff
@@ -112,6 +113,28 @@ func (r *ReconcileDatadogAgent) manageMetricsServerService(logger logr.Logger, d
 	return r.updateIfNeededMetricsServerService(logger, dda, service)
 }
 
+func (r *ReconcileDatadogAgent) manageMetricsServerAPIService(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
+	if dda.Spec.ClusterAgent == nil {
+		return r.cleanupMetricsServerAPIService(logger)
+	}
+
+	if !isMetricsProviderEnabled(dda.Spec.ClusterAgent) {
+		return reconcile.Result{}, nil
```

Comment:
Should't we also cleanup the APIService if the metrics provider is not enabled?

Also, `isMetricsProviderEnabled` already checks for a nil spec, so maybe those two conditionals can be just one that checks for `isMetricsProviderEnabled`.